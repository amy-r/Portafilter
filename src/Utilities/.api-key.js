export const key = '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'