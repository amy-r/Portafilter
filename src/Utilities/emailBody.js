import React from 'react'

export const body = (
  `<div>
    <h1> Hello, Amy! </h1>
    <p> 
      Thank you for subscribing to our email services!
      Below, you will find your roast review
    </p>
    <strong> 
      If at any time you'd like to unsubscribe,
      please click <a href='/'> Here </a> 
    </strong>
  </div>`
)

export const text = 'Thank you for subscribing to our email services! Below, you will find your roast review If at any time you\'d like to unsubscribe, please click here'