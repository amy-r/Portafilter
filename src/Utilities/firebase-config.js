import * as firebase from "firebase";

export const config = {
    apiKey: "AIzaSyBYOjbvNWIvFDu-9R6gR4PcBqkYt5H98g4",
    authDomain: "roastr-e35c3.firebaseapp.com",
    databaseURL: "https://roastr-e35c3.firebaseio.com",
    projectId: "roastr-e35c3",
    storageBucket: "roastr-e35c3.appspot.com",
    messagingSenderId: "112340188249"
};

export const uiConfig = {
  // Popup signin flow rather than redirect flow.
  signInFlow: 'popup',
  // Redirect to /signedIn after sign in is successful. Alternatively you can provide a callbacks.signInSuccess function.
  signInSuccessUrl: '/',
  // We will display Google and Facebook as auth providers.
  signInOptions: [
    firebase.auth.GoogleAuthProvider.PROVIDER_ID,
  ]
};
