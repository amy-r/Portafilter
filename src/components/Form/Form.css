@import url('https://fonts.googleapis.com/css?family=Open+Sans:300');


h1 {
  font-family: 'Open Sans', sans-serif;
  font-weight: 300;
  color: #444444;
  letter-spacing: 0.2rem;
}

.full, 
.half {
  display: block;
  color: #AA9E91;
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  font-weight: 300;
  margin: auto;
  border-style: none;
  padding: 5px;
  margin-top: 5px;
  height: 20px;
}

.full {
  width: 400px;
  margin: auto;
  margin-top: 15px;
}

.half {
  width: 186px;
  display: inline-block;
  margin-top: 15px;
}

.right {
  margin-left: 15px; 
}

.new-roaster {
  background-color: #BB997F;
  margin: auto;
  margin-top: 30px;
  width: 500px;
  padding: 20px;
  -webkit-box-shadow: 6px 4px 34px -5px rgba(0,0,0,0.75);
  -moz-box-shadow: 6px 4px 34px -5px rgba(0,0,0,0.75);
  box-shadow: 6px 4px 34px -5px rgba(0,0,0,0.75);
}

.submit {
  background-color: #444444;
  color: white;
  padding: 10px;
  margin: auto;
  margin-top: 25px;
  height: 50px;
  width: 80px;
  display: block;
  border-style: none;
}

.submit:hover {
  background-color: #898078;
}