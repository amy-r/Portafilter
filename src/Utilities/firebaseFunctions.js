import * as firebase from 'firebase';
import { config } from './firebase-config';
export const firebaseApp = firebase.initializeApp(config);

export const writeUserData = ({userId, userName, userEmail}) => {
  firebaseApp.database().ref('users/' + userId).set({
    username: userName,
    email: userEmail,
  });
}

export const addRoasterData = ({userId, name, location, altitude, equipment, water, contact, email}) => {
  firebaseApp.database().ref('roasters/' + name).set({
    name,
    altitude,
    location,
    equipment,
    water,
    contact,
    email,
  })
}

export const addCoffeeData = ({roaster, name, overallScore, region, acidity, body, sweetness, tactile, overallImpression, additionalComments}) => {
  firebaseApp.database().ref('coffees/' + name).set({
    roaster,
    overallScore,
    region,
    acidity,
    body,
    sweetness,
    tactile,
    overallImpression,
    additionalComments
  })
}

export const convertToArray = (roasters) => {
  return Object.keys(roasters).map( roaster => {
    return {...roasters[roaster]}
  });
}


export const pullRoasters = async (ref) => {
  const currentValue = await ref.once("value")
  const snapshotValue = await currentValue.val()
  return convertToArray(snapshotValue)
}
