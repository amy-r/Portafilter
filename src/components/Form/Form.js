import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addRoaster } from '../../actions/index';
import { withRouter } from 'react-router';
import './Form.css';
import { addRoasterData } from '../../Utilities/firebaseFunctions'

export class Form extends Component {
  constructor(props) {  
    super(props);
    this.state = {
      name:'',
      location: '',
      equipment: '',
      water: '',
      contact: '',
    }
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { history, user, addRoaster } = this.props
    const {name, location, equipment, water, contact} = this.state;
    const newRoaster = {
      userId: user.userId,
      name,
      location,
      equipment,
      water,
      contact
    } 
    addRoaster(newRoaster);
    addRoasterData(newRoaster);
    this.setState({
      name: '',
      location: '',
      equipment: '',
      water: '',
      contact: '',
    });
    history.push('/current-roasters');
  }

  handleChange = (event) => {
    const {name, value} = event.target
    this.setState({[name]:value})
  }

  render() {

    return (
      <form onSubmit= {this.handleSubmit}>
        <input type="text" 
          name="name" 
          value={this.state.name}
          placeholder='Name' 
          onChange={this.handleChange}/>
        <input type="text" 
          name="location" 
          value={this.state.location} 
          placeholder='Location'
          onChange={this.handleChange}/>
        <input type="text" 
          name="equipment" 
          value={this.state.equipment} 
          placeholder='Equipment'
          onChange={this.handleChange}/>
        <input type="text" 
          name="water" 
          value={this.state.water} 
          placeholder='Water'
          onChange={this.handleChange}/>
        <input type="text" 
          name="contact" 
          value={this.state.contact} 
          placeholder='Contact'
          onChange={this.handleChange}/>
        <input type="submit"/>
      </form>
    )
  }
}

export const mapStateToProps = state => ({
  user: state.user
})

export const mapDispatchToProps = dispatch => ({
  addRoaster: roaster => dispatch(addRoaster(roaster))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Form))
