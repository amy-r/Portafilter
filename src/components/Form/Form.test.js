/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import { Form, mapStateToProps, mapDispatchToProps } from './Form';
import { shallow } from 'enzyme';
import * as firebaseFunction from '../../Utilities/firebaseFunctions';
require('jest-localstorage-mock');

describe('Form', () => {
  it('should exist', () => {
    const wrapper = shallow(<Form />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should start with a state of empty strings', () => {
    const expectedState = {
      name:'',
      location: '',
      altitude: '',
      equipment: '',
      water: '',
      contact: '',
      email: '',
      errorState: ''
    }

    const wrapper = shallow(<Form />);

    expect(wrapper.state()).toEqual(expectedState);
  });

  it('should update state if the name field is typed into', () => {
    const expectedState = {
      name:'Amy',
      location: '',
      altitude: '',
      equipment: '',
      water: '',
      contact: '',
      email: '',
      errorState: ''
    }
    const wrapper = shallow(<Form />)
    const mockEvent = {
      target: {
        name: 'name',
        value: 'Amy'
    }}

    wrapper.find('input').first().simulate('change', mockEvent);
    expect(wrapper.state()).toEqual(expectedState);
  })

  it('should update state of every input if they are modified', () => {
    const expectedState = {
      name:'1',
      location: '1',
      altitude: '1',
      equipment: '1',
      water: '1',
      contact: '1',
      email: '1',
      errorState: ''
    }

    const wrapper = shallow(<Form />);

    const inputs = wrapper.find('input');
    const mockEventCreator = (inputName) => {
      return { 
        target: {
          name: inputName,
          value: '1'
        } 
      }
    };

    inputs.forEach( input => {
      input.simulate('change', mockEventCreator(input.prop('name')))
    });

    expect(wrapper.state()).toEqual(expectedState)
  });

  it('should set an error state if addRoasterData rejects', () => {
    firebaseFunction.addRoasterData = () => {
      throw new Error
    }
    const user = {userId: '1'};
    const mockEvent = {preventDefault: jest.fn()};
    const wrapper = shallow(<Form user={user} addRoaster={jest.fn()}/>);
    localStorage.setItem('roasters', JSON.stringify(['Corvus']) )

    wrapper.instance().handleSubmit(mockEvent);
    expect(wrapper.state('errorState')).toEqual('error adding roaster')
  })

  it('calls addRoaster on a successful submission', () => {
    firebaseFunction.addRoasterData = jest.fn();
    const user = {userId: '1'};
    const mockEvent = {preventDefault: jest.fn()};
    const wrapper = shallow(<Form user={user} addRoaster={jest.fn()}/>);
    localStorage.setItem('roasters', JSON.stringify(['Corvus']) )

    wrapper.instance().handleSubmit(mockEvent);

    expect(wrapper.props().addRoaster).toHaveBeenCalled;
  })
})

describe('MSTP and MDTP', () => {

  it('should define the user props for the container MSTP', () => {
    const userName= "Frank Ocean";
    const userEmail= "frank@gmail.com";
    const mockStore = {user: [ {name, userEmail} ]};
    const expected = [ {name, userEmail} ];
    const mapped = mapStateToProps(mockStore);
    expect(mapped.user).toEqual(expected)
  })
  
  it('should call dispatch on addRoaster', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addRoaster();
    expect(mockDispatch).toHaveBeenCalled;
  })
})