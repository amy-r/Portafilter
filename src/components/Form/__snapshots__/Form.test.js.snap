// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should exist 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
           ADD A NEW ROASTER 
</h1>,
        <form
          className="new-roaster"
          onSubmit={[Function]}
>
          <input
                    className="full"
                    name="name"
                    onChange={[Function]}
                    placeholder="NAME"
                    type="text"
                    value=""
          />
          <input
                    className="half"
                    name="location"
                    onChange={[Function]}
                    placeholder="LOCATION"
                    type="text"
                    value=""
          />
          <input
                    className="half right"
                    name="altitude"
                    onChange={[Function]}
                    placeholder="ALTITUDE"
                    type="text"
                    value=""
          />
          <input
                    className="full"
                    name="equipment"
                    onChange={[Function]}
                    placeholder="EQUIPMENT"
                    type="text"
                    value=""
          />
          <input
                    className="full"
                    name="water"
                    onChange={[Function]}
                    placeholder="WATER TDDS"
                    type="text"
                    value=""
          />
          <input
                    className="half"
                    name="contact"
                    onChange={[Function]}
                    placeholder="CONTACT"
                    type="text"
                    value=""
          />
          <input
                    className="half right"
                    name="email"
                    onChange={[Function]}
                    placeholder="EMAIL"
                    type="text"
                    value=""
          />
          <input
                    className="submit"
                    type="submit"
                    value="SUBMIT"
          />
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " ADD A NEW ROASTER ",
        },
        "ref": null,
        "rendered": " ADD A NEW ROASTER ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="full"
              name="name"
              onChange={[Function]}
              placeholder="NAME"
              type="text"
              value=""
/>,
            <input
              className="half"
              name="location"
              onChange={[Function]}
              placeholder="LOCATION"
              type="text"
              value=""
/>,
            <input
              className="half right"
              name="altitude"
              onChange={[Function]}
              placeholder="ALTITUDE"
              type="text"
              value=""
/>,
            <input
              className="full"
              name="equipment"
              onChange={[Function]}
              placeholder="EQUIPMENT"
              type="text"
              value=""
/>,
            <input
              className="full"
              name="water"
              onChange={[Function]}
              placeholder="WATER TDDS"
              type="text"
              value=""
/>,
            <input
              className="half"
              name="contact"
              onChange={[Function]}
              placeholder="CONTACT"
              type="text"
              value=""
/>,
            <input
              className="half right"
              name="email"
              onChange={[Function]}
              placeholder="EMAIL"
              type="text"
              value=""
/>,
            <input
              className="submit"
              type="submit"
              value="SUBMIT"
/>,
          ],
          "className": "new-roaster",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "full",
              "name": "name",
              "onChange": [Function],
              "placeholder": "NAME",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "half",
              "name": "location",
              "onChange": [Function],
              "placeholder": "LOCATION",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "half right",
              "name": "altitude",
              "onChange": [Function],
              "placeholder": "ALTITUDE",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "full",
              "name": "equipment",
              "onChange": [Function],
              "placeholder": "EQUIPMENT",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "full",
              "name": "water",
              "onChange": [Function],
              "placeholder": "WATER TDDS",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "half",
              "name": "contact",
              "onChange": [Function],
              "placeholder": "CONTACT",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "half right",
              "name": "email",
              "onChange": [Function],
              "placeholder": "EMAIL",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "submit",
              "type": "submit",
              "value": "SUBMIT",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             ADD A NEW ROASTER 
</h1>,
          <form
            className="new-roaster"
            onSubmit={[Function]}
>
            <input
                        className="full"
                        name="name"
                        onChange={[Function]}
                        placeholder="NAME"
                        type="text"
                        value=""
            />
            <input
                        className="half"
                        name="location"
                        onChange={[Function]}
                        placeholder="LOCATION"
                        type="text"
                        value=""
            />
            <input
                        className="half right"
                        name="altitude"
                        onChange={[Function]}
                        placeholder="ALTITUDE"
                        type="text"
                        value=""
            />
            <input
                        className="full"
                        name="equipment"
                        onChange={[Function]}
                        placeholder="EQUIPMENT"
                        type="text"
                        value=""
            />
            <input
                        className="full"
                        name="water"
                        onChange={[Function]}
                        placeholder="WATER TDDS"
                        type="text"
                        value=""
            />
            <input
                        className="half"
                        name="contact"
                        onChange={[Function]}
                        placeholder="CONTACT"
                        type="text"
                        value=""
            />
            <input
                        className="half right"
                        name="email"
                        onChange={[Function]}
                        placeholder="EMAIL"
                        type="text"
                        value=""
            />
            <input
                        className="submit"
                        type="submit"
                        value="SUBMIT"
            />
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ADD A NEW ROASTER ",
          },
          "ref": null,
          "rendered": " ADD A NEW ROASTER ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="full"
                name="name"
                onChange={[Function]}
                placeholder="NAME"
                type="text"
                value=""
/>,
              <input
                className="half"
                name="location"
                onChange={[Function]}
                placeholder="LOCATION"
                type="text"
                value=""
/>,
              <input
                className="half right"
                name="altitude"
                onChange={[Function]}
                placeholder="ALTITUDE"
                type="text"
                value=""
/>,
              <input
                className="full"
                name="equipment"
                onChange={[Function]}
                placeholder="EQUIPMENT"
                type="text"
                value=""
/>,
              <input
                className="full"
                name="water"
                onChange={[Function]}
                placeholder="WATER TDDS"
                type="text"
                value=""
/>,
              <input
                className="half"
                name="contact"
                onChange={[Function]}
                placeholder="CONTACT"
                type="text"
                value=""
/>,
              <input
                className="half right"
                name="email"
                onChange={[Function]}
                placeholder="EMAIL"
                type="text"
                value=""
/>,
              <input
                className="submit"
                type="submit"
                value="SUBMIT"
/>,
            ],
            "className": "new-roaster",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "full",
                "name": "name",
                "onChange": [Function],
                "placeholder": "NAME",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "half",
                "name": "location",
                "onChange": [Function],
                "placeholder": "LOCATION",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "half right",
                "name": "altitude",
                "onChange": [Function],
                "placeholder": "ALTITUDE",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "full",
                "name": "equipment",
                "onChange": [Function],
                "placeholder": "EQUIPMENT",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "full",
                "name": "water",
                "onChange": [Function],
                "placeholder": "WATER TDDS",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "half",
                "name": "contact",
                "onChange": [Function],
                "placeholder": "CONTACT",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "half right",
                "name": "email",
                "onChange": [Function],
                "placeholder": "EMAIL",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "submit",
                "type": "submit",
                "value": "SUBMIT",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
